collect_app/src/test/java/org/odk/collect/android/widgets/RangePickerDecimalWidgetTest.java

```diff
@@ -57,71 +67,76 @@ public void whenRangeStartIsGreaterThanRangeEnd_widgetsSetsCorrectValuesForNumbe
-        assertThat(widget.binding.widgetAnswerText.getText(), equalTo(widget.getContext().getString(R.string.no_value_selected)));
+        assertEquals(widget.binding.widgetAnswerText.getText(), widget.getContext().getString(R.string.no_value_selected));
+        assertEquals(widget.binding.widgetButton.getText(), widget.getContext().getString(R.string.select_value));
     }
 
     @Test
     public void clearAnswer_callsValueChangeListener() {
         RangePickerDecimalWidget widget = createWidget(promptWithQuestionDefAndAnswer(rangeQuestion, null));
         WidgetValueChangedListener valueChangedListener = mockValueChangedListener(widget);
-
         widget.clearAnswer();
+
         verify(valueChangedListener).widgetValueChanged(widget);
     }
 
     @Test
-    public void setNumberPickerValue_updatesAnswer() {
+    public void clickingWidgetForLong_callsLongClickListener() {
+        View.OnLongClickListener listener = mock(View.OnLongClickListener.class);
+
         RangePickerDecimalWidget widget = createWidget(promptWithQuestionDefAndAnswer(rangeQuestion, null));
-        widget.setNumberPickerValue(4);
+        widget.setOnLongClickListener(listener);
+        widget.binding.widgetButton.performLongClick();
+        widget.binding.widgetAnswerText.performLongClick();
 
-        assertThat(widget.getAnswer().getDisplayText(), equalTo("3.5"));
+        verify(listener).onLongClick(widget.binding.widgetButton);
+        verify(listener).onLongClick(widget.binding.widgetAnswerText);
     }
 
     @Test
-    public void setNumberPickerValue_whenRangeStartIsGreaterThenRangeEnd_updatesAnswer() {
-        when(rangeQuestion.getRangeStart()).thenReturn(new BigDecimal("5.5"));
-        when(rangeQuestion.getRangeEnd()).thenReturn(new BigDecimal("1.5"));
-
+    public void updatingValueInRangePickerViewModel_updatesAnswer() {
         RangePickerDecimalWidget widget = createWidget(promptWithQuestionDefAndAnswer(rangeQuestion, null));
         widget.setNumberPickerValue(4);
         assertThat(widget.getAnswer().getDisplayText(), equalTo("3.5"));
     }
 
     @Test
-    public void clickingWidgetForLong_callsLongClickListener() {
-        View.OnLongClickListener listener = mock(View.OnLongClickListener.class);
-
+    public void updatingValueInRangePickerViewModel_callsValueChangeListener() {
         RangePickerDecimalWidget widget = createWidget(promptWithQuestionDefAndAnswer(rangeQuestion, null));
-        widget.setOnLongClickListener(listener);
-        widget.binding.widgetButton.performLongClick();
-        widget.binding.widgetAnswerText.performLongClick();
+        WidgetValueChangedListener valueChangedListener = mockValueChangedListener(widget);
+        rangePickerViewModel.setNumberPickerValue(5);
 
-        verify(listener).onLongClick(widget.binding.widgetButton);
-        verify(listener).onLongClick(widget.binding.widgetAnswerText);
+        rangePickerViewModel.getNumberPickerValue().observe(fakeLifecycleOwner, integer -> {
```

Comment:
There's no reason for this to do an `observe`. You should just be able to call `verify(valueChangedListener).widgetValueChanged(widget);` after setting the value.